pyproject.toml

```diff
@@ -0,0 +1,31 @@
+[build-system]
+requires = [
+    "cmake>=3.19",
+    "numpy>=1.21.2",
+    "scikit-build-core >=0.4.3",
+    "ninja; platform_system!='Windows'"
+]
+
+build-backend = "scikit_build_core.build"
+
+[project]
+name = "finufft"
+version = "0.0.1"
```

Comment:
This and some of the other fields need to be migrated from `setup.py`. Also, for the version, it may be good to have this provided dynamically, but I'm not sure how to do that. Maybe something to look at later.

Reply:
sure. version can be taken from VCS: https://scikit-build-core.readthedocs.io/en/stable/configuration.html#dynamic-metadata

Reply:
Yeah I saw that, but I don't know that we're that fastidious about tagging versions properly in git (also it's not clear to me how it would handle things like `2.3.0dev0`). Easiest would be if it could pull it from one of the source files (like `__version__` in `finufft/__init__.py`).

Reply:
Ended up fixing this with a regex.